### PR TITLE
US1530 cStor volume (ZVOL) replica resize tasks

### PR DESCRIPTION
**Pull request by jkryl**
This is the second part of the fix which implements resize command
for management connection.

### CODE REVIEW

File: cmd/zrepl/mgmt_conn.c [hunk 1]
```diff
@@ -860,21 +879,58 @@ process_message(uzfs_mgmt_conn_t *conn)
...
+	case ZVOL_OPCODE_RESIZE:
+		if (payload_size != sizeof (*resize_data)) {
+			rc = reply_nodata(conn, ZVOL_OP_STATUS_FAILED,
+			    hdrp->opcode, hdrp->io_seq);
+			break;
+		}
+		resize_data = payload;
+
+		/* ref will be released when async command has finished */
+		if ((zinfo = uzfs_zinfo_lookup(resize_data->volname)) == NULL) {
```
Comment (vishnuitta) [inline cmd/zrepl/mgmt_conn.c:916]: having a function which does sanity check on conn will help?
not required for this PR.

File: include/zrepl_prot.h [hunk 2]
```diff
@@ -158,6 +159,14 @@ struct zrepl_status_ack {
 } __attribute__((packed));
 
 typedef struct zrepl_status_ack zrepl_status_ack_t;
+
+struct zvol_op_resize_data {
+	char	volname[MAX_NAME_LEN];	/* zvol to resize */
+	uint64_t size;			/* new size of zvol */
```
Comment (vishnuitta) [inline include/zrepl_prot.h:165]: 'size to 'volsize' may be good. Again, not needed in this PR.

**Review by vishnuitta — APPROVED** [review]
changes are good.

File: cmd/zrepl/mgmt_conn.c (hunk 1 above)
Comment (jkryl) [reply cmd/zrepl/mgmt_conn.c:916]: indeed there is too much repeating code. We will need to have a look at it in future.

**Pull request merged**